### ISSUE TITLE
[VPS-166](5) Creation of backend for managing scene permissions

### ISSUE THREAD

**Issue by JordanBlenn** [UoaWDCC/VPS#166]
**Description:**

**AS A:** Scenario Player
**I WANT:** to not be able to access scenes I should not have access to
**SO THAT:** I can fully experience the collaborative experience that is the virtual patient simulator.

**AS A:** Scenario Editor
**I WANT:** my choices around who can access what scenes to be saved
**SO THAT:** I can come back to previous scenes I chose roles for already and the choices are saved

**AC's:**
1) When I select the role(s) for a specific scene and the changes are saved, this information is saved in the backend
2) If I close and re-open the scene in editing more, my selections are still there.
3)  If, as a player I am playing through the scenario and reach a scene I cannot play through, it shows the "Invalid Scene Permissions" UI from #164 (with the correct role to play through also showing)

**Technical Discussion:**
- Saving in the backend should probably be done through giving the 'scene' class an attribute of "Allowed Roles", which can then be checked against the User's current role as they attempt to play through the scenario, as well as being pulled back to the UI to update which roles are allowed to play what scene
- blocked by #154, #164  and #165  (creating roles and Scene Permissions UI)
- AC 3 will require re-editing the code from #164 as it was implemented without any backend functionality
- Notes functionality is out of scope for this story
